presto-spi/src/main/java/com/facebook/presto/spi/block/SingleRowBlock.java

```diff
@@ -55,7 +55,7 @@ public long getSizeInBytes()
     {
         long sizeInBytes = 0;
         for (int i = 0; i < fieldBlocks.length; i++) {
-            sizeInBytes += getRawFieldBlock(i).getSizeInBytes();
+            sizeInBytes += getRawFieldBlock(i).getRegionSizeInBytes(rowIndex, 1);
```

Comment:
Want to check my understanding of what this change is about: A SingleRowBlock has a row index, and a set of fieldBlocks that have entries of the field more than one row. getSizeInBytes would get the size of the entire fieldBlock, but we want only the size of the entry for this row.  Is that right?  what about getRetainedSizeInBytes? does that also need to be changed?

Reply:
> A SingleRowBlock has a row index, and a set of fieldBlocks that have entries of the field more than one row. getSizeInBytes would get the size of the entire fieldBlock, but we want only the size of the entry for this row. Is that right?

Yes, that is correct.

> what about getRetainedSizeInBytes? does that also need to be changed?

My understanding is that `getRetainedSizeInBytes` is a number of bytes this object "holds"(retains) inmemory. In other words this is a number of bytes that otherwise would've been garbage collected. Do i understand that correctly, @nezihyigitbasi ?